cumulusci/tasks/bulkdata/load.py

```diff
@@ -107,56 +108,43 @@ def _load_mapping(self, mapping):
         if "RecordTypeId" in mapping["fields"]:
             conn = self.session.connection()
             self._load_record_types([mapping["sf_object"]], conn)
-        mapping["oid_as_pk"] = bool(mapping.get("fields", {}).get("Id"))
-        job_id, local_ids_for_batch = self._create_job(mapping)
-        result = self._wait_for_job(job_id, error_behaviour="return")
-
-        self._process_job_results(mapping, job_id, local_ids_for_batch)
-
-        return result
 
-    def _create_job(self, mapping):
-        """Initiate a bulk insert or update and upload batches to run in parallel."""
-        action = mapping["action"]
-        step_mode = mapping.get("bulk_mode")
-        task_mode = self.bulk_mode
-        mode = step_mode or task_mode or "Parallel"
+        mapping["oid_as_pk"] = bool(mapping.get("fields", {}).get("Id"))
 
-        if action == "insert":
-            job_id = self.bulk.create_insert_job(
-                mapping["sf_object"], contentType="CSV", concurrency=mode
-            )
-        else:
-            job_id = self.bulk.create_update_job(
-                mapping["sf_object"], contentType="CSV", concurrency=mode
-            )
+        step = BulkApiDmlOperation(
+            sobject=mapping["sf_object"],
+            operation=(
+                DataOperationType.INSERT
+                if mapping.get("action") == "insert"
+                else DataOperationType.UPDATE
+            ),
+            api_options={"bulk_mode": self.bulk_mode or "Parallel"},
```

Comment:
bulk_mode was previously set from the mapping. That code seems to have been deleted. Was that deliberate?